<launch>

  <arg name="robot_name" default=""/>
  <arg name="x"/>
  <arg name="y"/>


  <group ns="$(arg robot_name)">
    
    <node pkg="map_merging" type="map_merging" respawn="false" name="map_merge" output="screen">
      <param name="merged_map_topic" value="merged_map"/>
      <param name="world_frame" value="/world"/>
      <param name="pose_topic" value="geometry_pose"/>
      <param name="map_topic" value="map"/>
      <param name="max_comm_distance" value="100.0"/>
      <param name="max_number_robots" value="5"/>
      <param name="merging_rate" value="10"/>
      <param name="init_pose_x"  value="$(arg x)"/>
      <param name="init_pose_y"  value="$(arg y)"/>
      <param name="init_pose_z"  value="0"/>
      <param name="init_pose_yaw"  value="0"/>
      <param name="robot_base_frame" value="$(arg robot_name)"/>
    </node>

    <!-- <node pkg="tf" type="static_transform_publisher" name="world_to_map_tf_broadcaster"  args="$(arg x) $(arg y) 0 0 0 0 /world $(arg robot_name)/map 100"/>  -->

    <node pkg="tf2_ros" type="static_transform_publisher" name="world_to_$(arg robot_name)_tf_broadcaster" args="$(arg x) $(arg y) 0 0 0 0 /world /$(arg robot_name)/map 100"/>


    <node pkg="costmap_2d" type="costmap_2d_node" name="costmap_for_exploration_global" respawn="true">
      
      <rosparam ns="costmap" subst_value="true">
        <!-- footprint: [[-0.5, -0.33], [-0.5, 0.33], [0.5, 0.33], [0.5, -0.33]]
        footprint_padding: 0.01 -->
        global_frame: $(arg robot_name)/map
        robot_base_frame: $(arg robot_name)/base_link
        update_frequency: 1
        publish_frequency: 1
        rolling_window: false
        always_send_full_costmap: true
        # Map management parameters
        unknown_cost_value: 255
        lethal_cost_threshold: 55
        <!-- trinary_costmap: true -->
        <!-- obstacle_layer: {enabled: true, lethal_cost_threshold: 100, track_unknown_space: true} -->
        robot_radius: 0.3
        static:
          #Can pull data from gmapping, map_server or a non-rolling costmap            
          map_topic: $(arg robot_name)/merged_map
          subscribe_to_updates: true 
        inflation:
          inflation_radius: 0.5
          cost_scaling_factor: 10

        plugins: 
          - {name: static, type: "costmap_2d::StaticLayer"}
          - {name: inflation,        type: "costmap_2d::InflationLayer"}  
        

      </rosparam>
      </node>
  </group>
  
</launch>
